app/models/transformation_mapping_item.rb

```diff
@@ -24,4 +31,33 @@ def destination_cluster
       errors.add(:destination, "EMS type of destination cluster or cloud tenant must be in: #{destination_types}")
     end
   end
+
+  def source_datastore
+    tm                   = TransformationMapping.find(self.transformation_mapping_id)
+    tmis                 = tm.transformation_mapping_items.where(:source_type => "EmsCluster")
+    src_cluster_storages = tmis.collect(&:source).collect(&:storages).flatten
+    source_storage       = self.source
```

Comment:
This `self` should not be needed

Reply:
Yes I will be removing all `self`. I was playing with it for understand what is what.